org.eclipse.xtext/src/org/eclipse/xtext/workspace/UnknownProjectConfig.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.workspace;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+
+public class UnknownProjectConfig implements IProjectConfig {
+	private final String name = "__unknown_project";
+
+	private final IWorkspaceConfig workspaceConfig;
+
+	@Override
+	public ISourceFolder findSourceFolderContaining(final URI member) {
+		return null;
+	}
+
+	@Override
+	public URI getPath() {
+		return null;
+	}
+
+	@Override
+	public Set<? extends ISourceFolder> getSourceFolders() {
+		return Collections.emptySet();
+	}
+
+	public UnknownProjectConfig(final IWorkspaceConfig workspaceConfig) {
```

Comment:
The order of the members is weird. Constructors do usually appear at the top after the fields throughout our codebase.

Reply:
done